src/tests/ftest/pool/evict_test.py

```diff
@@ -38,110 +39,249 @@ class EvictTest(TestWithServers):
+        self.pool.create()
+        self.log.info("Valid Pool UUID is %s", self.pool.uuid)
 
-        try:
-            create_cmd = (
-                "{0} create-pool "
-                "-m {1} "
-                "-u {2} "
-                "-g {3} "
-                "-s {4} "
-                "-c 1".format(self.daosctl, mode, uid, gid, setid))
-            uuid_str = """{0}""".format(process.system_output(create_cmd))
-            print("uuid is {0}\n".format(uuid_str))
-
-        except Exception as excep:
-            print(excep)
-            print(traceback.format_exc())
-            self.fail("Unexpected pool create and connect failure.\n")
-        try:
-            # use the wrong uuid, which of course should fail
-            bogus_uuid = "44be695840f4b6eda581b461a4d4c570490ba4db"
-            bad_evict_cmd = (
-                "{0} evict-pool "
-                "-i {1} "
-                "-s {2}".format(self.daosctl, bogus_uuid, setid))
-            process.system(bad_evict_cmd)
-
-        except Exception as excep:
-            # this section of the test should fail and throw an exception
-            print("command expected to fail")
+        # Check that the pool was created
+        self.assertTrue(
+            self.pool.check_files(self.hostlist_servers),
+            "Pool data not detected on servers before evict")
+
+        # Connect to the pool
+        self.assertTrue(self.pool.connect(1),
+                        "Pool connect failed before evict")
+        self.log.info(
+            "Pool UUID is %s and handle is %s",
+            self.pool.uuid, self.pool.pool.handle.value)
 
+        # Attempt to evict pool with invald UUID
+        saved_uuid = self.pool.uuid
+        bogus_uuid = '81ef94d7-a59d-4a5e-935b-abfbd12f2105'
+        # set the UUID directly
+        self.pool.pool.set_uuid_str(bogus_uuid)
+        self.log.info("Evicting pool with Invalid Pool UUID:  %s",
+                      self.pool.pool.get_uuid_str())
         try:
-            # use the wrong server group name but the correct uuid
-            bogus_server_group = self.params.get("setname",
-                                                 '/run/tests/setnames/'
-                                                 'badsetname/*')
-            bad_evict_cmd = (
-                "{0} evict-pool "
-                "-i {1} "
-                "-s {2}".format(self.daosctl, uuid_str, bogus_server_group))
-            process.system(bad_evict_cmd)
-
-        except Exception as excep:
-            # this section of the test should fail and throw an exception
-            print("command expected to fail")
+            self.log.info("Attempting to evict pool with an invalid UUID")
+            # call daos evict api directly
+            self.pool.pool.evict()
+        # exception is expected
+        except DaosApiError as result:
+            self.log.info(
+                "Expected exception - evicting pool with invalid UUID\n %s",
```

Comment:
Can we test here for the exact expected exception? Say for example if the exception returns ERR -1001 then make sure we are getting that one and not something else.